RTR build.py tool does not work with modern lcov versions

|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.14.4 | CI | Manager | Sources | Ubuntu 24.04 |

## Description

The build system and CI scripts (build.py, run_check.py) fail when running on Ubuntu 24.04 LTS. These same commands work correctly on Ubuntu 22.04 LTS.

The failures are caused by newer versions of the development toolchain included in Ubuntu 24.04, specifically:
- Cppcheck (2.13+): Scans the build/ directory recursively and fails on temporary CMake compiler check files.
- LCOV (2.0+): Changed its output format (removed dots ......), breaking the regex parser in run_check.py.

## Steps to reproduce
On an Ubuntu 24.04 environment follow general build script instrutions for TARGET=server and run a module build, for example:
```sh
python3 build.py -r shared_modules/dbsync --cpus 4
```

## Result
The build fails at different stages.
#### Cppcheck Failure
Cppcheck analyzes Cmake temporary files inside the `build` directory treating compiler macros as syntax errors.
```sh
/home/vagrant/wazuh/src/wazuh_modules/syscollector/build/CMakeFiles/3.28.3/CompilerIdC/CMakeCCompilerId.c:757:28: error: syntax error [syntaxError]
char const* info_version = "INFO" ":" "compiler_version[" COMPILER_VERSION "]";
```
#### Coverage/LCOV Failure
After fixing the previous failure, the coverage check crashes python because the regex expects the old LCOV format (which uses '.' instead of spaces). The current regexp matching in `ci/run_check.py:checkCoverage` is strict and fails with this new format:
```sh
ValueError: could not convert string to float: ''
```

--- a/src/ci/run_check.py
+++ b/src/ci/run_check.py
@@ -31,28 +31,25 @@ def checkCoverage(output):
     Raises:
         - ValueError: Raises an exception when fails for some reason.
     """
-    reLines = re.search("lines.*(% ).*(lines)", str(output))
-    reFunctions = re.search("functions.*%", str(output))
-    if reLines:
-        end = reLines.group().index('%')
-        start = reLines.group()[0:end].rindex(' ') + 1
-        linesCoverage = reLines.group()[start:end]
-    if reFunctions:
-        end = reFunctions.group().index('%')
-        start = reFunctions.group().rindex(' ') + 1
-        functionsCoverage = reFunctions.group()[start:end]
+    output_str = str(output)
+    matchLines = re.search(r"lines[\.\s]*:\s+([\d\.]+)\%", output_str)
+    matchFunctions = re.search(r"functions[\.\s]*:\s+([\d\.]+)\%", output_str)
+    linesCoverage = matchLines.group(1) if matchLines else "0.0"
+    functionsCoverage = matchFunctions.group(1) if matchFunctions else "0.0"
+
     if float(linesCoverage) >= 90.0:
-        utils.printGreen(msg="[Lines Coverage {}%: PASSED]"
-                         .format(linesCoverage))
+        utils.printGreen(msg="[Lines Coverage {}%: PASSED]".format(linesCoverage))
     else:
         utils.printFail(msg="[Lines Coverage {}%: LOW]".format(linesCoverage))
         errorString = "Low lines coverage: {}".format(linesCoverage)
         raise ValueError(errorString)
+
     if float(functionsCoverage) >= 90.0:
-        utils.printGreen(msg="[Functions Coverage {}%: PASSED]"
-                         .format(functionsCoverage))
+        utils.printGreen(
+            msg="[Functions Coverage {}%: PASSED]".format(functionsCoverage)
+        )
     else:
-        utils.printFail(msg="[Functions Coverage {functionsCoverage}%: LOW]")
+        utils.printFail(msg="[Functions Coverage {}%: LOW]".format(functionsCoverage))
         errorString = "Low functions coverage: {}".format(functionsCoverage)
         raise ValueError(errorString)
 
@@ -273,7 +270,7 @@ def runCppCheck(moduleName):
                       headerKey="cppcheck")
 
     currentDir = utils.moduleDirPath(moduleName)
-    cppcheckCommand = "cppcheck --force --std=c++14 --quiet {}".format(currentDir)
+    cppcheckCommand = "cppcheck --force --std=c++17 --quiet -i {}/build {}".format(currentDir, currentDir)
 
     out = subprocess.run(cppcheckCommand,
                          stdout=subprocess.PIPE,

